programs/drift/src/instructions/user.rs

```diff
@@ -1281,18 +1269,27 @@ pub fn handle_place_and_make_spot_order<'info>(
     }
 
     let market_index = params.market_index;
-    let mut fulfillment_params = match fulfillment_type {
+
+    let mut fulfillment_params: Box<dyn SpotFulfillmentParams> = match fulfillment_type {
```

Comment:
Do we need the `Box` here? Does cargo get angry if you just use something like `&'a mut dyn SpotFulfillmentParams`

Reply:
the compiler complained, if you can get it to work without the box i'd be happy to do that!